agents/check_mk_agent.macosx

```diff
@@ -84,66 +249,296 @@ netstat -ntfinet | awk ' /^tcp/ { c[$6]++; } END { for (x in c) { print x, c[x];
 
 
 # Fileinfo-Check: put patterns for files into /etc/check_mk/fileinfo.cfg
-perl -e '
-use File::Glob "bsd_glob";
-my @patterns = ();
-foreach (bsd_glob("$ARGV[0]/fileinfo.cfg"), bsd_glob("$ARGV[0]/fileinfo.d/*")) {
-    open my $handle, "<", $_ or next;
-    while (<$handle>) {
-        chomp;
-        next if /^\s*(#|$)/;
-        my $pattern = $_;
-        $pattern =~ s/\$DATE:(.*?)\$/substr(`date +"$1"`, 0, -1)/eg;
-        push @patterns, $pattern;
-    }
-    warn "error while reading $_: $!\n" if $!;
-    close $handle;
-}
-exit if ! @patterns;
-
-print "<<<fileinfo:sep(124)>>>\n", time, "\n[[[header]]]\nname|status|size|time\n[[[content]]]\n";
-
-foreach (@patterns) {
-    foreach (bsd_glob("$_")) {
-        if (! -f) {
-            print "$_|missing\n" if ! -d;
-        } elsif (my @infos = stat) {
-            print "$_|ok|$infos[7]|$infos[9]\n";
-        } else {
-            print "$_|stat failed: $!\n";
-        }
-    }
-}
-' -- "$MK_CONFDIR"
+if [ -r "$MK_CONFDIR/fileinfo.cfg" ] ; then
+    echo '<<<fileinfo:sep(124)>>>'
+    date +%s
+
+    OLD_IFS=$IFS
+    IFS='
+'
+    while read -r pattern; do
+        case $pattern in
+            /*) for file in $pattern; do
+                    stat -f "%N|%z|%m" "$file" 2>/dev/null || echo "$file|missing|`date +%s`"
+                done ;;
+        esac
+    done < "$MK_CONFDIR/fileinfo.cfg"
+    IFS=$OLD_IFS
+fi
 
+# temperatures and sensors, requires HardwareMonitor.app or osx-cpu-temp
+if [ -x /Applications/HardwareMonitor.app/Contents/MacOS/hwmonitor ]; then
+    echo '<<<lnx_thermal:sep(124)>>>'
+    /Applications/HardwareMonitor.app/Contents/MacOS/hwmonitor -c | grep " C$" | while read ; do
+        Temp=$(awk -F": " '{print $2}' <<<$REPLY | tr -d -c '[:digit:]')
+        case $REPLY in
+            SMART*)
+                Sensor="$(sed 's/^SMART Disk //' <<<"$REPLY" | cut -d'(' -f1)"
+                echo "${Sensor}|enabled|disk-thermal|${Temp}000|50000|passive|60000|critical"
+                ;;
+            "SMC DRIVE BAY"*)
+                Sensor="$(sed 's/^SM. //' <<<"$REPLY" | awk -F": " '{print $1}')"
+                echo "${Sensor}|enabled|disk-thermal|${Temp}000|50000|passive|60000|critical"
+                ;;
+            SMC*|SMB*)
+                Sensor="$(sed 's/^SM. //' <<<"$REPLY" | awk -F": " '{print $1}')"
+                echo "${Sensor}|enabled|thermal|${Temp}000|70000|passive|90000|critical"
+                ;;
+        esac
+    done
+elif type osx-cpu-temp >/dev/null
```

Comment:
According to Linux Agent, this line should be changed by inserting and using the `inpath()` function as in the Linux agent:
```
# Function to replace "if type [somecmd]" idiom
# 'command -v' tends to be more robust vs 'which' and 'type' based tests
inpath() {
    command -v "${1:?No command to test}" >/dev/null 2>&1
}
```